src/index.ts

```diff
@@ -63,11 +68,81 @@ function successLoader(
       : rawFilePath;
 
   const fileVersion = updateFileInCache(filePath, contents, instance);
+  const referencedProject = getProjectReferenceForFile(filePath, instance);
+  if (referencedProject) {
+    if (referencedProject.commandLine.options.outFile) {
+      throw new Error(
+        `The referenced project at ${
+          referencedProject.sourceFile.fileName
+        } is using the ` +
+          `outFile' option, which is not supported with ts-loader.`
+      );
+    }
 
-  const { outputText, sourceMapText } = options.transpileOnly
-    ? getTranspilationEmit(filePath, contents, instance, loader)
-    : getEmit(rawFilePath, filePath, instance, loader);
+    const jsFileName = instance.compiler.getOutputJavaScriptFileName(
+      filePath,
+      referencedProject.commandLine
+    );
+    const mapFileName = jsFileName + '.map';
+    if (!instance.compiler.sys.fileExists(jsFileName)) {
+      throw new Error(
+        `Could not find output JavaScript file for input ${filePath} ` +
+          `(looked at ${jsFileName}).\nThe input file is part of a project ` +
+          `reference located at ${referencedProject.sourceFile.fileName}, ` +
+          'so ts-loader is looking for the project’s pre-built output on ' +
+          'disk. Try running `tsc --build` to build project references.'
+      );
+    }
 
+    if (!instance.compiler.sys.fileExists(mapFileName)) {
+      loader.emitWarning(
```

Comment:
It might be obnoxious to do this for every file. Maybe I can memoize by `referencedProject.sourceFile.fileName`? Or would it make more sense to validate each project reference while first setting up the program/languageService/watch?

Reply:
That seems reasonable; presumably that's not something that changes often at all so should be a completely reasonable choice